python/pyspark/sql/connect/session.py

```diff
@@ -236,6 +237,38 @@ def __init__(self, connection: Union[str, ChannelBuilder], userId: Optional[str]
         self._client = SparkConnectClient(connection=connection, user_id=userId)
         self._session_id = self._client._session_id
 
+    @classmethod
+    def _set_default_and_active_session(cls, session: "SparkSession") -> None:
+        """
+        Set the (global) default :class:`SparkSession`, and (thread-local)
+        active :class:`SparkSession` when they are not set yet.
+        """
+        with cls._lock:
+            if cls._default_session is None:
+                cls._default_session = session
+        if getattr(cls._active_session, "session", None) is None:
+            cls._active_session.session = session
+
+    @classmethod
+    def getActiveSession(cls) -> Optional["SparkSession"]:
```

Comment:
Is this also a public API? I found it a bit confusing that this `.getActiveSession()` returns the active session but `.active()` returns the active or the default session (and throws an exception if no active session). Is the API the same as non Spark Connect one?

Reply:
yup